packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue

```diff
@@ -1,295 +1,191 @@
+			"following" option is not supported yet
+		</i>
+		<header class="flex w-full gap-3 mb-2">
+			<tera-toggleable-input
+				:model-value="config.name"
+				tag="h3"
+				@update:model-value="emit('update-self', { key: 'name', value: $event })"
 			/>
-		</div>
-
-		<div class="section-row">
-			<div class="button-row">
-				<label>Constraint type</label>
-				<Dropdown
-					:model-value="constraintType"
-					:options="constraintTypes"
-					option-value="id"
-					option-label="name"
-					placeholder="Select constraint type"
-					@update:model-value="changeConstraintType($event)"
+			<div class="ml-auto flex align-items-center">
+				<label class="mr-2">Active</label>
+				<InputSwitch
+					class="mr-3"
+					:model-value="config.isActive"
+					:disabled="
+						// Lock if constraintType is Following
+						config.constraintType === ConstraintType.Following
+					"
+					@update:model-value="emit('update-self', { key: 'isActive', value: $event })"
 				/>
+				<Button icon="pi pi-trash" text rounded @click="emit('delete-self')" />
 			</div>
-
-			<div class="button-row">
-				<label>Target</label>
-				<MultiSelect
-					v-if="constraintType !== 'parameterConstraint'"
-					v-model="variables"
-					:options="props.modelStates"
-					placeholder="Model states"
-					display="chip"
-					@update:model-value="updateChanges()"
-				></MultiSelect>
-				<MultiSelect
-					v-else
-					v-model="variables"
-					:options="props.modelParameters"
-					placeholder="Model states"
-					display="chip"
-					@update:model-value="updateChanges()"
-				></MultiSelect>
-			</div>
-		</div>
-
-		<!-- Weights -->
-		<div v-for="(variable, index) of variables" :key="index">
-			<div class="button-row">
-				<label v-if="weights">
-					{{ variable + ' Weight' }}
-				</label>
+		</header>
+		<p>
+			The
+			<Dropdown
+				:model-value="config.constraint"
+				:options="Object.values(Constraint)"
+				@update:model-value="emit('update-self', { key: 'constraint', value: $event })"
+			/>
+			<MultiSelect
+				:model-value="config.variables"
+				placeholder="Select variables"
+				:options="variableOptions"
+				@update:model-value="emit('update-self', { key: 'variables', value: $event })"
+			/>
+			should be
+			<Dropdown
+				:model-value="config.constraintType"
+				:options="Object.values(ConstraintType)"
+				@update:model-value="
+					($event: ConstraintType) => {
+						// Disable isActive if constraintType is Following
+						if ($event === ConstraintType.Following) {
+							emit('update-self', { key: 'isActive', value: false });
+						}
+						emit('update-self', { key: 'constraintType', value: $event });
+					}
+				"
+			/>
+			<tera-input-number
+				v-if="config.constraintType === ConstraintType.LessThan"
+				auto-width
+				:model-value="config.interval?.ub ?? 0"
+				@update:model-value="emit('update-self', { key: 'interval', value: { lb: config.interval?.lb, ub: $event } })"
```

Comment:
If `config.intevel.ub` is not defined, you would set the model-value as a literal `0`, then for update you would still emit `config.intervel.ub` which is still undefined no?

The setup seems weird here.

Reply:
It changes to `0` as soon as you click out of the input. This is not right. Once I activate this input `config.interval.ub` should be `0`